mlprimitives/candidates/keras_LSTM_CycleGAN.py

```diff
@@ -0,0 +1,276 @@
+# -*- coding: utf-8 -*-
+
+import logging
+from functools import partial
+
+import keras
+import numpy as np
+from keras import backend as K
+from keras.layers import (
+    LSTM, Activation, Bidirectional, Conv1D, Dense, Dropout, Flatten, Input, Reshape,
+    TimeDistributed)
+from keras.layers.advanced_activations import LeakyReLU
+from keras.layers.convolutional import UpSampling1D
+from keras.layers.merge import _Merge
+from keras.models import Model
+from keras.optimizers import Adam
+
+LOGGER = logging.getLogger(__name__)
+
+
+class RandomWeightedAverage(_Merge):
+    def _merge_function(self, inputs):
+        alpha = K.random_uniform((64, 1, 1))
+        return (alpha * inputs[0]) + ((1 - alpha) * inputs[1])
+
+
+class LSTM_CycleGAN():
+    """LSTM CycleGAN class"""
+
+    def build_encoder(self):
+        x = Input(shape=self.shape)
+        model = keras.models.Sequential()
+        model.add(Bidirectional(LSTM(100, return_sequences=True)))
+        model.add(Flatten())
+        model.add(Dense(self.latent_dim))
+        model.add(Reshape((self.latent_dim, 1)))
+        z = model(x)
+        return Model(x, z)
+
+    def build_generator(self):
+        z = Input(shape=(self.latent_dim, 1))
+        model = keras.models.Sequential()
+        model.add(Flatten())
+        model.add(Dense(50))
+        model.add(Reshape((50, 1)))
+        model.add(
+            Bidirectional(
+                LSTM(
+                    64,
+                    return_sequences=True,
+                    dropout=0.2,
+                    recurrent_dropout=0.2),
+                merge_mode='concat'))
+        model.add(UpSampling1D(2))
+        model.add(
+            Bidirectional(
+                LSTM(
+                    64,
+                    return_sequences=True,
+                    dropout=0.2,
+                    recurrent_dropout=0.2),
+                merge_mode='concat'))
+        model.add(TimeDistributed(Dense(1)))
+        model.add(Activation("tanh"))
+        x_ = model(z)
+        return Model(z, x_)
+
+    def build_critic_x(self):
+        x = Input(shape=(self.shape[0], 1))
```

Comment:
Perhaps it would be interesting to make all these arguments configurable as inputs.

I think that we could follow a similar approach to what the `keras.Sequential` adapter does, reading the network architectures form the input arguments, so one can calibrate and adapt the network to their needs if necessary.

Optionally, we can move the `mlprimitives.adapters.keras.Sequential._build_model` method out from the class and make it a function, so each model can be built here by calling that function and passing it the corresponding network architecture.